Feature/blog

Mateaus opened the pull request:
Added content:
- Post list UI
- Post content UI
- Post models
- Next and Previous content button
- Blog url routing connected with main project.
- CSS update with custom variables for font-size and font color.
Bug fixed:
- Navbar currentview couldn't find the current active link after opening a post. Now it only looks for one if there is one available.

Mateaus committed:
Created blank blog view and connected the route.

Mateaus committed:
Sample UI added for blog

Mateaus committed:
Created models and UI for the blog view.

Mateaus committed:
Added blog post view with a simple ui.

Mateaus committed:
Added and improved the post content view.

Mateaus committed:
Cleaned and improved css and html blog files.

Mateaus committed:
Improved post_list and post_view htmls, improved css font to scale based on screen size.

Mateaus committed:
Added extra padding for next/prev buttons inside post details view.

Mateaus committed:
Fixed extra padding happening in the post list.

Mateaus committed:
Fixed blog container top padding. Switched margin for padding for inner content.

Mateaus committed:
Added color for all headers and paragraphs to inherit from root color variable.

Mateaus committed:
Fixed a bug in the navbar js file.

Anthony9698 commented on personalwebsite/static/blog/dist/js/blog.js:
```diff
@@ -0,0 +1,21 @@
+
+function textBlock() {
+    var truncateAtIndex = 200;
+    var text = document.getElementsByClassName("text-block");
```
Where exactly is this being used? I searched all files for the "text-block" class and couldn't find anything.

Anthony9698 commented on personalwebsite/templates/base_page_view.html:
```diff
@@ -35,7 +36,7 @@
 
             <a id="my-current-view" class="navbar-brand font-medium"
-                style="margin: 0 0 0 0; padding-left: 16px; display: none;" href="">Navbar</a>
+                style="margin: 0 0 0 0; padding-left: 16px; display: none;" href=""></a>
```
This style should probably be in its own class.

Anthony9698 commented on personalwebsite/blog/templates/blog/post_detail.html:
```diff
@@ -0,0 +1,151 @@
+
+    <hr>
+
+    <div class="post-content">{{ post_detail.content|safe }}</div>
```
Word wrapping should be enabled here for long post detail content. I think you can add this to your .post-content class.

Anthony9698 commented on personalwebsite/blog/models.py:
```diff
@@ -1,3 +1,33 @@
+    title = models.CharField(max_length=256)
+    overview = models.TextField()
+    tags = models.ManyToManyField(Tag)
+    estimated_time = models.IntegerField()
```
I think in a future task we can automate this by using some sort of API that estimates reading time-based on word count or make our own. For now, this is fine though,

Mateaus replied:
That is a good idea, I will add it to the list.

Anthony9698 reviewed (changes requested):
This is a really good addition to the site and I really like the way everything is set up so far. I only have a few minor changes that I would make before merging with main.

Mateaus committed:
Fixed ui bug where text in post_detail wasn't wrapping.

Mateaus committed:
Improved navbar and fixed a bug with the post overview not wrapping.

Mateaus committed:
Merge branch 'main' into feature/blog
Updating current branch with the most up to date main branch.

Mateaus committed:
Updated migrations

Mateaus commented:
Changes:
- Fixed bug where post overview wasn't wrapping if the text was too long.
- Improved navbar, removed unnecessary js and instead now is mostly being handled by css.
